server/task/src/main/java/org/apache/james/task/MemoryTaskManager.java

```diff
@@ -144,7 +144,13 @@ public TaskExecutionDetails await(TaskId id) {
                 .filter(ignore -> tasksResult.get(id) != null)
                 .map(ignore -> {
                     Optional.ofNullable(tasksResult.get(id))
-                        .ifPresent(Throwing.<Mono<Task.Result>>consumer(Mono::block).orDoNothing());
+                        .ifPresent(mono -> {
```

Comment:
how is it supposed to change anything ?

Reply:
`CancellationException` is a runtime exception...

Here is the code of `orDoNothing`...

```
    public Consumer<T> orDoNothing()
    {
        return t -> {
            try {
                throwing.doAccept(t);
            } catch (Error | RuntimeException e) {
                throw e;
            } catch (Throwable ignored) {
                // nothing
            }
        };
    }
```

Obviously in case of cancellation, this was not doing what you expects....

Reply:
great, thank you